src/content/reference/react/useTransition.md

```diff
@@ -1672,13 +2138,17 @@ startTransition(async () => {
 However, this works instead:
 
 ```js
-await someAsyncFunction();
-startTransition(() => {
-  // ✅ Setting state *during* startTransition call
-  setPage('/about');
+startTransition(async () => {
+  await someAsyncFunction();
+  // ✅ Using startTransition *after* await
+  startTransition(() => {
+    setPage('/about');
+  });
```

Comment:
Something that isn't immediately clear to me:

Do we need this nested `startTransition` for:
- regular form actions too?
- calling `setOptimisticValue`?

I assume we do, but not 100% sure since I don't see any example in the docs.

The `<form>` actions examples are likely to use `useActionState` instead of doing `startTransition(() => setState())`

The `useOptimistic` doc only calls the setter as the first step, probably the most common case, but maybe it can make sense in some rare cases to call it after `await`? I don't know 🤷‍♂️

```tsx
  async function formAction(formData) {
    addOptimisticMessage(formData.get("message"));
    formRef.current.reset();
    await sendMessage(formData);
  }
```

Reply:
Form actions will call that out if needed on the form page, the examples there will not all use useActionState, but even if they do you would need to add the startTransition after the await there too. For useOptimistic, you could have multiple awaits, and want to call it between the two.